### DISCUSSION TITLE
TTS RESTful API得到的base64文件解析问题求助

### DISCUSSION THREAD

**Discussion by centurion2021** [PaddlePaddle/PaddleSpeech#2060] [Q&A] [answered]
Hello，
如题，调用RESTful API生成的base64 decode后播放声音变形严重，但是服务器端保存下来的wav文件却是好的。最开始怀疑是自己转换base64的方法有问题，结果发现转换wav文件的base64值和API返回的base64不一样，完全没头绪了。
测试代码如下
`
import base64
import json
import requests
import numpy as np
import sounddevice as sd

request_data = {
    "text": '你好',
"spk_id": 0,
"speed": 1.0,
"volume": 1.0,
'sample_rate': 16000,
"save_path": './nihao.wav'
}

r = requests.post('http://192.168.1.5:8090/paddlespeech/tts', data=json.dumps(request_data)).json()
print(r)

base64_audio_data = r['result']['audio']
audio_data = np.frombuffer(buffer=base64.b64decode(base64_audio_data), dtype=np.int16)

sd.play(audio_data, samplerate=16000, blocking=True)`

**Comment by lym0302** [comment, marked as answer]
1、API返回的base64转音频可以参考下述链接对应的函数：https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/paddlespeech/server/bin/paddlespeech_client.py#L84 

2、转换wav文件的base64值和API返回的base64不一样 是因为 API返回的base64并不是来源于保存的 wav文件，因为考虑到服务端有可能并不想保存音频的情况。参考下述链接 174~192行：https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/paddlespeech/server/engine/tts/python/tts_engine.py#L174
